### PR TITLE
Integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by osbentech**
In this branch I implemented the following:
- solved the N+1 problem when fetching all posts and their comments for a user.
before
image

after
image

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -5,6 +5,6 @@ def index
```
Comment (gemmen29) [inline app/controllers/users_controller.rb:5]: - [ ] Kindly fix the `n + 1` problem that exists in the [view](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_user_all_posts.png) given at requirement which exists in post controller at the index action.
### Hint: you can use the `include` method to fetch all comments on the post.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @osbentech, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -1,6 +1,6 @@
-  <div class="posts">
+ <div class="posts">
     <%= render partial: "shared/user", locals: { user: @user } %>
-    <% @posts.each do |post| %>
+    <% @posts.includes(:comments).each do |post| %>
```
Comment (anagudelogu) [inline app/views/posts/index.html.erb:3]: - [ ] **[OPTIONAL]** Great job working on the previous reviewer's suggestions! 🚀 However, I would suggest that you move this `:includes` to your controller as it is meant to be part of the query. Applying this change will make your code more readable and better organized by following the Single Responsibility principle in which views are just responsible for displaying the data provided by the controller! 🤓 🔧 


```rb
# app/controllers/posts_controller.rb
class PostsController < ApplicationController
  ...
  def index
    @user = User.find(params[:user_id])
    @posts = @user.posts.includes(:comments).order(created_at: :asc)
  end
  ...
end
```

**Review by anagudelogu — APPROVED** [review]
# STATUS: APPROVED! 🍾 🎆 🥇 

Hi @osbentech, 
I'm Andres.

You did a great job working on this milestone, your project is complete! 🚀 
There is nothing else to say other than... it's time to merge it 🤖  
Congratulations! 🎉 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**